synapse/rest/client/v2_alpha/devices.py

```diff
@@ -248,10 +261,19 @@ def __init__(self, hs):
         self.device_handler = hs.get_device_handler()
 
     async def on_POST(self, request: SynapseRequest):
-        requester = await self.auth.get_user_by_req(request, allow_guest=True)
+        requester = await self.auth.get_user_by_req(request)
 
         submission = parse_json_object_from_request(request)
 
+        if "device_id" not in submission:
+            raise errors.SynapseError(
+                400, "device_id missing", errcode=errors.Codes.MISSING_PARAM,
+            )
+        elif isinstance(submission["device_id"], str):
```

Comment:
I think this is missing a `not`:

```suggestion
        elif not isinstance(submission["device_id"], str):
```